Content.Server/Nyanotrasen/Item/PseudoItem/PseudoItemSystem.cs

```diff
@@ -1,174 +1,43 @@
-    // {
-    //     if (component.Active)
-    //         args.Cancel();
-    // }
-    //
-    // private void OnDoAfter(EntityUid uid, PseudoItemComponent component, DoAfterEvent args)
-    // {
-    //     if (args.Handled || args.Cancelled || args.Args.Used == null)
-    //         return;
-    //
-    //     args.Handled = TryInsert(args.Args.Used.Value, uid, component);
-    // }
-    //
-    // public bool TryInsert(EntityUid storageUid, EntityUid toInsert, PseudoItemComponent component,
-    //     StorageComponent? storage = null)
-    // {
-    //     if (!Resolve(storageUid, ref storage))
-    //         return false;
-    //
-    //     if (component.Size > storage.StorageCapacityMax - storage.StorageUsed)
-    //         return false;
-    //
-    //     var item = EnsureComp<ItemComponent>(toInsert);
-    //     _tagSystem.TryAddTag(toInsert, PreventTag);
-    //     _itemSystem.SetSize(toInsert, component.Size, item);
-    //
-    //     if (!_storageSystem.Insert(storageUid, toInsert, out _, null, storage))
-    //     {
-    //         component.Active = false;
-    //         RemComp<ItemComponent>(toInsert);
-    //         return false;
-    //     }
-    //
-    //     component.Active = true;
-    //     return true;
-    // }
-    //
-    // private void StartInsertDoAfter(EntityUid inserter, EntityUid toInsert, EntityUid storageEntity,
-    //     PseudoItemComponent? pseudoItem = null)
-    // {
-    //     if (!Resolve(toInsert, ref pseudoItem))
-    //         return;
-    //
-    //     var ev = new PseudoItemInsertDoAfterEvent();
-    //     var args = new DoAfterArgs(EntityManager, inserter, 5f, ev, toInsert, toInsert, storageEntity)
-    //     {
-    //         BreakOnTargetMove = true,
-    //         BreakOnUserMove = true,
-    //         NeedHand = true
-    //     };
-    //
-    //     _doAfter.TryStartDoAfter(args);
-    // }
-    //
-    // private void OnInsertAttempt(EntityUid uid, PseudoItemComponent component,
-    //     ContainerGettingInsertedAttemptEvent args)
-    // {
-    //     if (!component.Active)
-    //         return;
-    //     // This hopefully shouldn't trigger, but this is a failsafe just in case so we dont bluespace them cats
-    //     args.Cancel();
-    // }
+    public override void Initialize()
+    {
+        SubscribeLocalEvent<PseudoItemComponent, GetVerbsEvent<AlternativeVerb>>(AddInsertAltVerb);
+    }
+
+    // For whatever reason, I have to put these in server or the verbs duplicate
+    private void AddInsertAltVerb(EntityUid uid, PseudoItemComponent component, GetVerbsEvent<AlternativeVerb> args)
+    {
+        if (!args.CanInteract || !args.CanAccess)
+            return;
+
+        if (component.Active)
+            return;
+
+        if (!TryComp<StorageComponent>(args.Using, out _))
+            return;
+
+        // There *should* be a check here to see if we can fit, but I'm not aware of an easy way to do that, so eh, who cares
```

Comment:
SharedStorageSystem.CanInsert might work

Reply:
That requires you to have an item. Felinids aren't considered an item until past that. I could adapt the check, but I'll do that tomorrow

Reply:
I rewrote the entire fucking check (or well, copied big chunks of it and replaced stuff here and there) to work with pseudoitem, not fun but it works